Gems/ROS2/Code/Source/Transform/ROS2FrameComponent.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include "TransformPublisher.h"
+#include <AzCore/Component/Component.h>
+#include <AzCore/std/smart_ptr/unique_ptr.h>
+
+namespace ROS2
+{
+    /// This component marks an interesting reference frame and captures common behavior
+    /// for sensor data frame of reference, publishing ros2 static and dynamic transforms (tf)
+    /// and helping to export robot prefab to URDF.
+    class ROS2FrameComponent
+        : public AZ::Component
+    {
+    public:
+        AZ_COMPONENT(ROS2SensorComponent, "{EE743472-3E25-41EA-961B-14096AC1D66F}", AZ::Component);
+
+        void Activate() override;
+        void Deactivate() override;
+
+        static void Reflect(AZ::ReflectContext* context);
+        static void GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided);
+        static void GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible);
+        static void GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required);
+
+        // namespaced and validated, ready to send in ros2 messages (TODO-validate)
+        AZStd::string GetFrameID() const;
+        AZStd::string GetNamespace() const;
+
+        // Transform for direct parent if parent ROS2Frame is found, otherwise transformation is global ("world")
+        const AZ::Transform& GetFrameTransform() const;
+
+    private:
+        AZ::TransformInterface* GetEntityTransformInterface() const;
+        const ROS2FrameComponent* GetParentROS2FrameComponent() const;
+
+        // If parent entity does not exist or does not have a ROS2Frame component, return ros2 default global frame: "world"
+        AZStd::string GetParentFrameID() const;
```

Comment:
If `GetParentROS2FrameComponent()` would always return non-null, this could be removed and replaced with `GetParentROS2FrameComponent()->GetFrameID()`.

Reply:
See https://github.com/RobotecAI/o3de/pull/8#discussion_r850633139